app/src/components/SignUpForm.js

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react';
+import {Validations} from '../common/Validations.js';
+import UsersDataRequests from '../common/UsersDataRequests';
+import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import '../css/App.css';
+import {connect} from 'react-redux';
+import '../../node_modules/jquery/dist/jquery.min'
+import UserForm from './UserForm';
+import isGuest from "../common/isGuest";
+import endPointsList from "../common/endPointsList";
+import {NavLink} from 'react-router-dom'
+import Alerts from './Alerts';
+
+class SignUpForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+
+        this.signUpInputId = {
+            firstName: 'sign-up-firstName',
+            lastName: 'sign-up-lastName',
+            email: 'sign-up-email',
+            genderMale: 'sign-up-gender-male',
+            genderFemale: 'sign-up-gender-female',
+            age: 'sign-up-age',
+            photo: 'sign-up-photo',
+            middleName: 'sign-up-middleName'
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    async handleSubmit(event) {
+        event.preventDefault();
+        const validateFlag = this.validateForm();
+        if (validateFlag) {
+            console.log('valid validation');
+            const result = await UsersDataRequests.signUpReq(this.props.signup);
+
+            this.setState({...this.state, alert: result});
+        } else {
+            console.log('invalid validation');
+        }
+    }
+    
+    validateForm() {
+        const validStatus = document.getElementById('valid-status');
+        const validateFlag = Validations.validateForm(this.props.signup);
+        if (validateFlag) {
+            validStatus.innerHTML = '';
+            validStatus.classList.remove('alert-danger');
+            const middleName = this.props.signup.middleName;
+            if (!middleName.isValid) {
```

Comment:
In case if I have a lot of optional fields, you should implement an approach which gives the opportunity to maintain many optional fields without the pain.